### PR TITLE
Overheating - Add stats to arsenal

### PR DESCRIPTION
**Pull request by Drofseh**
**When merged this pull request will:**
- Adds an arsenal stat to show if the weapon is open or closed bolt.
- Adds an arsenal stat to show if the weapon's barrel can be swapped.

Baer lied to me and changing the branch name closed #9717 (or maybe I screwed up the name change lol)

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/overheating/ACE_Arsenal_Stats.hpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
```
Comment (PabstMirror) [inline addons/overheating/ACE_Arsenal_Stats.hpp:1]:
```suggestion
```

File: addons/overheating/ACE_Arsenal_Stats.hpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class EGVAR(arsenal,stats) {
+    class statBase;
+    class ACE_allowSwapBarrel: statBase {
```
Comment (LinkIsGrim) [inline addons/overheating/ACE_Arsenal_Stats.hpp:3]: Would add condition property here with constant text property instead. IMO this only needs to be shown if the weapon actually supports barrel swapping

File: addons/overheating/functions/fnc_statTextStatement_allowSwapBarrel.sqf [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#include "..\script_component.hpp"
+/*
+ * Author: drofseh
+ * Bolt Type statement.
```
Comment (PabstMirror) [inline addons/overheating/functions/fnc_statTextStatement_allowSwapBarrel.sqf:4]:
```suggestion
 * Barrel Type statement.
```

**Review by PabstMirror — APPROVED** [review]
lgtm

File: addons/overheating/ACE_Arsenal_Stats.hpp (hunk 2 above)
Comment (Drofseh) [reply addons/overheating/ACE_Arsenal_Stats.hpp:3]: I'd like it to show if they can't be swapped too. For example the PKP is an upgraded PKM and if my community that's resulted in people assuming you can swap the barrel when it's actually not possible to do on the PKP.

**Review by LinkIsGrim — APPROVED** [review]

**Pull request merged**